run_web.py

```diff
@@ -0,0 +1,4254 @@
+        # Print data import message for each ECM if in verbose mode
+        verboseprint("Imported ECM '" + m.name + "' competition data")
+
+    # Import total absolute heating and cooling energy use data, used in
+    # removing overlaps between supply-side and demand-side heating/cooling
+    # ECMs in the analysis
+    with open(path.join(base_dir, *handyfiles.htcl_totals), "r") as msi:
+        try:
+            htcl_totals = json.load(msi)
+        except ValueError as e:
+            raise ValueError(
+                "Error reading in '" + handyfiles.htcl_totals + "': " + str(e)
+            ) from None
+
+    # Print message to console; if in verbose mode, print to new line,
+    # otherwise append to existing message on the console
+    if options.verbose:
+        print("ECM competition data load complete")
+    else:
+        print("Data load complete")
+
+    # Instantiate an Engine object using active measures list
+    a_run = Engine(handyvars, measures_objlist, energy_out)
+
+    # Calculate uncompeted and competed measure savings and financial
+    # metrics, and write key outputs to JSON file
+    for adopt_scheme in handyvars.adopt_schemes:
+        # Calculate each measure's uncompeted savings and metrics,
+        # and print progress update to user
+        print(
+            "Calculating uncompeted '" + adopt_scheme + "' savings/metrics...", end="", flush=True,
+        )
+        a_run.calc_savings_metrics(adopt_scheme, "uncompeted")
+        print("Calculations complete")
+        # Update each measure's competed markets to reflect the
+        # removal of savings overlaps with competing measures,
+        # and print progress update to user
+        print("Competing ECMs for '" + adopt_scheme + "' scenario...", end="", flush=True)
+        a_run.compete_measures(adopt_scheme, htcl_totals)
+        print("Competition complete")
+        # Calculate each measure's competed measure savings and metrics
+        # using updated competed markets, and print progress update to user
+        print(
+            "Calculating competed '" + adopt_scheme + "' savings/metrics...", end="", flush=True,
+        )
+        a_run.calc_savings_metrics(adopt_scheme, "competed")
+        print("Calculations complete")
+        # Write selected outputs to a summary JSON file for post-processing
+        a_run.finalize_outputs(adopt_scheme)
+
+    # Notify user that all analysis engine calculations are completed
+    print("All calculations complete; writing output data...", end="", flush=True)
+
+    # Write results to output files
+    if options.web is True:
+        # Write results dict to the ecm_results column in the analysis table
+        try:
+            # Ensure that the analysis_ecms match the --ids argument
+            query = "DELETE FROM " + db["schema"] + ".analysis_ecms WHERE analysis_id = %s"
+            cursor.execute(query, options.analysis_id)
```

Comment:
```suggestion
            cursor.execute(query, (options.analysis_id,))
```
Without this change, this query will fail when the analysis_id has multiple digits.